File: master-coding-interview/sorting/insertionSort.js
//efficient when the list is almost sorted
function insertionSort(array){
    const len = array.length
    for(let i = 1; i < len; i++){
        let current_val = array[i]
        let index = i
        while(index > 0 && array[index - 1] > current_val){
            array[index] = array[index - 1]
            index --;
        }
        array[index] = current_val
    }
    return array
}

File: sorting/insertion_sort.rb
def insertion_sort(input)
  input.each_with_index do |elem, i|
    index = i
    while index > 0 && input[index - 1] > elem do
      input[index] = input[index - 1]
      index -= 1
    end
    input[index] = elem
  end
  input
end